lib/models/update-checker.js

```diff
@@ -64,31 +64,29 @@ UpdateChecker.prototype.doCheck = function() {
 };
 
 UpdateChecker.prototype.checkNPM = function() {
-  // make an http call to npm to get the latest version
-  var http = require('http');
-  var concat = require('concat-stream');
+  // use npm to determine the currently availabe ember-cli version
+  var loadNPM = Promise.denodeify(require('npm').load);
+  var getEmberCliVersion = function(npm){
+    return Promise.denodeify(npm.commands.view)(['ember-cli', 'version']);
+  };
+  var extractVersion = function(data) { return Object.keys(data)[0]; };
 
-  return new Promise(function(resolve, reject) {
-    http.get('http://registry.npmjs.org/ember-cli/latest', function(res) {
-      res.setEncoding('utf8');
-
-      res.pipe(concat(function (data) {
-        try {
-          resolve(JSON.parse(data).version);
-        } catch(error) {
-          reject(error);
-        }
-      }));
-
-      res.on('error', reject);
-    });
-  }).then(function(version) {
-    // we only want to save the version information when we check NPM
-    return this.saveVersionInformation(version);
-  }.bind(this)).catch(function(error) {
-    this.ui.writeLine('There was an error checking NPM for an update: ' + error);
-    throw error;
-  }.bind(this));
+  return loadNPM({
+      'loglevel': 'silent',
+      'fetch-retries': 1,
+      'cache-min': 1,
+      'cache-max': 0
+    })
+    .then(getEmberCliVersion)
+    .then(extractVersion)
+    .then(function(version) {
+      // we only want to save the version information when we check NPM
+      return this.saveVersionInformation(version);
+    }.bind(this))
+    .catch(function(error) {
+      this.ui.writeLine('There was an error checking NPM for an update: ' + error);
```

Comment:
Why are we trying to catching those errors/rejects at the level https://github.com/xomaczar/ember-cli/blob/master/lib/models/update-checker.js#L86-L88.
The only way any of the those promises get rejected if an error gets thrown, we don't explicitly reject() anything in any then(...). Let's just https://github.com/ember-cli/ember-cli/blob/master/lib/cli/cli.js#L98 catch and log it.